### PR TITLE
Add CD part

### PR DESCRIPTION
**Pull request by jlevesy**
### What does this PR do?

This PR adds the `CD` chapter:
- It shows how to use github releases from github actions
- It shows how to do continous deployment with heroku

### CODE REVIEW

File: content/chapitres/cd.adoc [hunk 1]
```diff
@@ -36,8 +69,6 @@ Continuous Deployment
 
 * Version "avancée" de la livraison continue:
 ** Chaque changement *est* déployé en production, de manière *automatique*
-* Question importante: En avez-vous besoin ?
-** Avez-vous les mêmes besoin que Amazon Google ou Netflix ?
```
Comment (jlevesy) [inline content/chapitres/cd.adoc:40]: I highly disagree with that point: CD should be instated as soon as possible in a company life. It has so many benefits even for a small company. That's why I removed it.

File: content/chapitres/setup-dev-env.adoc [hunk 2]
```diff
@@ -40,7 +40,7 @@ Lors de votre 1er accès, GitPod va vous demander des autorisations (à accepter
 
 Cliquez sur le bouton ci-dessous pour démarrer un environnement GitPod personnalisé:
 
-image::https://gitpod.io/button/open-in-gitpod.svg[link="{gitpod_url}",window="_blank"]
+image::https://gitpod.io/button/open-in-gitpod.svg[link="https://gitpod.io#https://github.com/cicd-lectures/gitpod",window="_blank"]
```
Comment (jlevesy) [inline content/chapitres/setup-dev-env.adoc:43]: We need to create a new workspace at this stage. That's why I  put back the link.

File: content/chapitres/cd.adoc (hunk 1 above)
Comment (dduportal) [reply content/chapitres/cd.adoc:40]: We need to have a discussion on this topic, but let's keep it removed for now.

[edit]
I mean by this bullet that not everyone needs (and can afford) a production automatically upgraded on each (successful) commit: banking environements, legally enforced platforms and many other examples require human approval before production deployment, and cannot always afford a staging environment.


But you make me realize something: this bullet is badly phrased so better removing it and if i can explain better I'll  propose another one.

File: content/chapitres/cd.adoc [hunk 3]
```diff
@@ -28,6 +28,57 @@ ____
...
+
+Hello Github Releases!
+
+== Anatomie d'une Release GitHub
+
+* Une release est associée à un tag git
+* Porte un titre
+* Un descriptif des changements
+* Des tarballs du code source a cette version
+* Et une collection d'assets arbitraires (des binaires compilés par exemple)
```
Comment (dduportal) [inline content/chapitres/cd.adoc]:
```suggestion
Une release GiHub est associée à un tag git et porte :

* Un titre
* Un descriptif des changements
* Une collection de d'assets dont:
** Des tarballs du code source a cette version (automatique)
** Et éventuellement des fichiers de votre choix (des binaires compilés par exemple)
```

File: content/chapitres/cd.adoc [hunk 4]
```diff
@@ -28,6 +28,57 @@ ____
...
+
+image::menu-server-releases.png[]
+
+== Prérequis: Ordonancement et Execution Contionelle des Jobs
```
Comment (dduportal) [inline content/chapitres/cd.adoc]:
```suggestion
== Prérequis: Ordonnancement et exécution conditionnelle des jobs
```

File: content/chapitres/cd.adoc [hunk 5]
```diff
@@ -28,6 +28,57 @@ ____
...
+
+* Un workflow execute par défaut tous les jobs qui le compose en parallèle
```
Comment (dduportal) [inline content/chapitres/cd.adoc]:
```suggestion
* Un workflow exécute en parallèle, par défaut, tous les jobs le composant
```

File: content/chapitres/cd.adoc [hunk 6]
```diff
@@ -28,6 +28,57 @@ ____
...
+* Il est possible de conditoner l'execution d'un job à la terminaison d'un autre à l'aide du mot clé `needs` (link:https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#jobsjob_idneeds[doc])
```
Comment (dduportal) [inline content/chapitres/cd.adoc]:
```suggestion
* Il est possible de conditionner l’exécution d'un job à la terminaison d'un autre à l'aide du mot clé `needs` (link:https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#jobsjob_idneeds[documentation de `needs`])
** Exécution séquentielle au lieu de parallèle
```

File: content/chapitres/cd.adoc [hunk 7]
```diff
@@ -28,6 +28,57 @@ ____
...
+* Il est possible d'executer conditionellement un job ou un step à l'aide du mot clé `if` (link:https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#jobsjob_idif[doc])
```
Comment (dduportal) [inline content/chapitres/cd.adoc]:
```suggestion
* Il est possible d’exécuter conditionnellement un job ou un step à l'aide du mot clé `if` (link:https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#jobsjob_idif[documentation de `if`])
```

File: content/chapitres/tests.adoc [hunk 8]
```diff
@@ -157,7 +157,7 @@ public void listsKnownMenus() {
 
 == Exercice : Corriger un Bug (7/11)
 
-* Super on à un test, il ne reste plus qu'à le lancer avec `mvn run` 🎉
+* Super on à un test, il ne reste plus qu'à le lancer avec `mvn test` 🎉
```
Comment (dduportal) [inline content/chapitres/tests.adoc]:
```suggestion
* Super on a un test unitaire, il ne reste plus qu'à le lancer avec `mvn test` 🎉
```

File: content/chapitres/tests.adoc [hunk 9]
```diff
@@ -223,7 +223,7 @@ public void listsKnownMenus() {
 
 == Exercice : Corriger un Bug (10/11)
 
-* Super on à un test (qui teste!), il ne reste plus qu'à le lancer avec `mvn run` 🎉
+* Super on à un test (qui teste!), il ne reste plus qu'à le lancer avec `mvn test` 🎉
```
Comment (dduportal) [inline content/chapitres/tests.adoc]:
```suggestion
* Super on a un test unitaire (qui teste!), il ne reste plus qu'à le lancer avec `mvn test` 🎉
```

**Review by dduportal — COMMENTED** [review]
Great job!

First batch of review (typo/wording) only.

The structure looks good to me, still have to play it "end to end"

File: content/chapitres/cd.adoc [hunk 10]
```diff
@@ -46,6 +98,153 @@ image::continuous-depl-vs-delivery.jpg[caption=Continuous Delivery vs Deployment
...
+* Connection directe avec GitHub
+* Push d'une image dans une container registry
+** C'est cette méthode que nous allons utiliser (car la moins automagique)
+
+== Construire une Image de Container pour Heroku
+
+[source,Dockerfile]
+----
+# Depuis l'image de base eclipse-temurin:11 (qui embarque un JRE dans la version 11)
+FROM eclipse-temurin:11
```
Comment (jlevesy) [inline content/chapitres/cd.adoc:155]:
```suggestion
FROM azul/zulu-openjdk:11
```

**Review by dduportal — APPROVED** [review]
🚀

**Pull request merged**